modules/product/src/main/java/com/opengamma/strata/product/bond/FixedCouponBondOption.java

```diff
@@ -0,0 +1,800 @@
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.google.common.collect.ImmutableSet;
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.Resolvable;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.date.AdjustableDate;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.product.Product;
+import com.opengamma.strata.product.common.LongShort;
+
+/**
+ * An option on a {@link FixedCouponBond}.
+ * <p>
+ * The option strike is expressed as "Clean price". The "clean" price excludes any accrued interest. The clean price
+ * is in the currency of the underlying bond.
+ * <p>
+ * The call/put is provided by the quantity's sign. If positive, it indicates the right to buy the bond (call), 
+ * if negative it indicates the right to sell the bond (put).
+ */
+@BeanDefinition
+public final class FixedCouponBondOption
+    implements Product, Resolvable<ResolvedFixedCouponBondOption>, ImmutableBean, Serializable {
+
+  /**
+   * Whether the option is long or short.
+   * <p>
+   * Long indicates that the owner has the right to exercise the option at expiry.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LongShort longShort;
+  /**
+   * The bond underlying the option.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final FixedCouponBond underlying;
+  /**
+   * The expiry date of the option.
+   * <p>
+   * This is the last date that the option can be exercised.
+   * <p>
+   * This date is typically set to be a valid business day.
+   * However, the {@code businessDayAdjustment} property may be set to provide a rule for adjustment.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final AdjustableDate expiryDate;
+  /**
+   * The expiry time of the option.
+   * <p>
+   * The expiry time is related to the expiry date and time-zone.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LocalTime expiryTime;
+  /**
+   * The time-zone of the expiry time.
+   * <p>
+   * The expiry time-zone is related to the expiry date and time.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ZoneId expiryZone;
+  /**
+   * The quantity that was traded.
+   * <p>
+   * This will be positive if buying (call) and negative if selling (put).
+   */
+  @PropertyDefinition
+  private final double quantity;
+  /**
+   * The <i>clean</i> price at which the option can be exercised, in decimal form.
+   * <p>
+   * The "clean" price excludes any accrued interest.
+   * <p>
+   * Strata uses <i>decimal prices</i> for bonds in the trade model, pricers and market data.
+   * For example, a price of 99.32% is represented in Strata by 0.9932.
+   */
+  @PropertyDefinition(validate = "ArgChecker.notNegative")
+  private final double strike;
```

Comment:
worth naming this cleanStrikePrice? It would be clearer when dealing with the object higher up in the stack.

Reply:
name changed